### PR TITLE
CANS-604 - Summary of assessment ratings

### PR DESCRIPTION
**Pull request by wmitchell-cambria**
## Description

[CANS-604 - Summary of Assessment Ratings](https://osi-cwds.atlassian.net/browse/CANS-604)

This adds a summary table to the Assessment page, between the header and main assessment form, which displays for completed assessments. It contains a four lists of items, one for the child's strengths, one for trauma, and two for items needing attention or immediate attention. While the items in the list are displayed using internationalization, the domains that are considered as Strength and Trauma domains are currently hard coded. This is because we must support existing assessments. With versioning, we could support various different configurations, but that is a large task to tackle in the future.

Because behavior is not specified yet in the story, an item from a caregiver domain can show up twice, if it has a 2 or 3 rating from multiple different caregiver domains.

The story also requires tool tips, which are not included in this PR. Those will be in a follow-up PR soon.

Lastly, a lot of the items and titles are cut off right now (ellipsis...), because the side bar makes the page narrower than the design anticipated. The sidebar will be removed in another story, which will help rectify this. If we need to adjust the font-size or something in the meantime, we can do that.

## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by wmitchell-cambria** [comment]
Here is a screenshot of a (very large) summary:
screen shot 2018-11-27 at 12 41 25 pm

File: app/javascript/Application/components/Assessment/AssessmentSummary/DomainHelper.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    PropTypes.shape({
+      code: PropTypes.string.isRequired,
+      rating: PropTypes.number.isRequired,
+    })
+  ).isRequired,
+})
+
+export const DomainsPropType = PropTypes.arrayOf(DomainPropType)
+
+const STRENGTH_DOMAINS = ['STR', 'EST']
```
Comment (AK56) [inline app/javascript/Application/components/Assessment/AssessmentSummary/DomainHelper.js]: I am unclear about the 'EST' abbreviation for childStrengthItems

Comment (wmitchell-cambria) [reply app/javascript/Application/components/Assessment/AssessmentSummary/DomainHelper.js]: It is the early childhood (Age 0-5) strength domain.

**Comment by wmitchell-cambria** [comment]
Retest this please. (Jenkins looks healthier now.)

**Pull request merged**